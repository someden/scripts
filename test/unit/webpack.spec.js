describe('@trigen/scripts-preset-react-app', () => {
	describe.skip('webpack', () => {
		const {
			dev,
			build,
			render
		} = require('../../packages/scripts-preset-react-app/src/webpack');

		it('should place dev plugins in correct order', () => {
			const plugins = dev().plugins.map(_ => _.constructor.name);

			expect(plugins).toMatchSnapshot();
		});

		it('should place build plugins in correct order', () => {
			const plugins = build().plugins.map(_ => _.constructor.name);

			expect(plugins).toMatchSnapshot();
		});

		it('should place render plugins in correct order', () => {
			const plugins = render().plugins.map(_ => _.constructor.name);

			expect(plugins).toMatchSnapshot();
		});
	});
});
