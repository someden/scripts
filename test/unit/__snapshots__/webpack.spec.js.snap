// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@trigen/scripts-preset-react-app webpack should place build plugins in correct order 1`] = `
Array [
  "CleanPlugin",
  "HashedModuleIdsPlugin",
  "HtmlWebpackPlugin",
  "ScriptExtHtmlWebpackPlugin",
  "HtmlWebpackSkipAssetsPlugin",
  "EnvironmentPlugin",
  "DefinePlugin",
  "FilterWarningsPlugin",
  "CopyPlugin",
  "ForkTsCheckerWebpackPlugin",
  "WorkboxWebpackPlugin",
  "StylelintWebpackPlugin",
  "StylableWebpackPlugin",
  "FilterChunkWebpackPlugin",
]
`;

exports[`@trigen/scripts-preset-react-app webpack should place dev plugins in correct order 1`] = `
Array [
  "HotModuleReplacementFilterPlugin",
  "HotModuleReplacementPlugin",
  "HtmlWebpackPlugin",
  "EnvironmentPlugin",
  "DefinePlugin",
  "FilterWarningsPlugin",
  "CopyPlugin",
  "ForkTsCheckerWebpackPlugin",
  "WorkboxWebpackPlugin",
  "StylelintWebpackPlugin",
  "StylableWebpackPlugin",
]
`;

exports[`@trigen/scripts-preset-react-app webpack should place render plugins in correct order 1`] = `
Array [
  "CleanPlugin",
  "HashedModuleIdsPlugin",
  "EnvironmentPlugin",
  "DefinePlugin",
  "FilterWarningsPlugin",
  "CopyPlugin",
  "ForkTsCheckerWebpackPlugin",
  "WorkboxWebpackPlugin",
  "StylableWebpackPlugin",
]
`;
